pkg/controller/persistentvolume/persistentvolume_claim_binder_controller.go

```diff
@@ -166,6 +192,11 @@ func syncVolume(volumeIndex *persistentVolumeOrderedIndex, binderClient binderCl
 		volumeIndex.Add(volume)
 	}
 
+	if isBeingProvisioned(volume) {
+		glog.V(4).Infof("Skipping PersistentVolume[%s], waiting for provisioning to finish", volume.Name)
+		return nil
+	}
+
 	switch currentPhase {
```

Comment:
It feels like there will be race conditions between the binder and provisioner controllers since both operate on the same API objects without any apparent locking mechanism before them.